src/taxonomy/taxonomy-detail/TaxonomyDetailMenu.jsx

```diff
@@ -20,9 +20,16 @@ const TaxonomyDetailMenu = ({
       alt={intl.formatMessage(messages.actionsButtonAlt, { name })}
       disabled={disabled}
     >
-      <Dropdown.Item onClick={() => onClickMenuItem('export')}>
-        {intl.formatMessage(messages.exportMenu)}
-      </Dropdown.Item>
+      { menuItems.includes('export') && (
+        <Dropdown.Item onClick={() => onClickMenuItem('export')}>
+          {intl.formatMessage(messages.exportMenu)}
+        </Dropdown.Item>
+      )}
+      { menuItems.includes('delete') && (
+        <Dropdown.Item onClick={() => onClickMenuItem('delete')}>
+          {intl.formatMessage(messages.deleteMenu)}
+        </Dropdown.Item>
+      )}
```

Comment:
Likewise you can use map here as well.

Reply:
Updated here: https://github.com/openedx/frontend-app-course-authoring/pull/684/commits/a4fc2ac7d4c9f14aac2df119f0ca5e1b06a3a4e0